File: apps/flutter-app/lib/features/transactions/domain/entities/transaction.dart
import 'package:equatable/equatable.dart';

enum TransactionType { subscription, cancellation }

class FundTransaction extends Equatable {
  final int id;
  final int fundId;
  final String fundName;
  final String category;
  final TransactionType type;
  final double amount;
  final String notification;
  final DateTime createdAt;

  const FundTransaction({
    required this.id,
    required this.fundId,
    required this.fundName,
    required this.category,
    required this.type,
    required this.amount,
    required this.notification,
    required this.createdAt,
  });

  /// Whether this is a subscription transaction.
  bool get isSubscription => type == TransactionType.subscription;

  /// Whether this is a cancellation transaction.
  bool get isCancellation => type == TransactionType.cancellation;

  @override
  List<Object?> get props => [
        id,
        fundId,
        fundName,
        category,
        type,
        amount,
        notification,
        createdAt,
      ];
}

File: packages/shared/api-contract/src/models/transaction.ts
import { NotificationMethod } from '../enums/notification-method';

export type TransactionType = 'subscription' | 'cancellation';

export interface Transaction {
  readonly id: string;
  readonly fundId: string;
  readonly fundName: string;
  readonly type: TransactionType;
  readonly amount: number;
  readonly date: string;
  readonly notificationMethod?: NotificationMethod;
}
